pkg/validation/spec/ref.go

```diff
@@ -72,7 +72,7 @@ func (r *Ref) IsValidURI(basepaths ...string) bool {
 		if err != nil {
 			return false
 		}
-
+		rr.Body.Close()
```

Comment:
thanks for the leak fix... it looks like IsValidURI is unused in this repo or in kubernetes/kubernetes, and I don't think we actually *want* code paths that can be triggered to make network requests from here ... I'm wondering if we should delete IsValidURI entirely